### PR TITLE
fix(assembly): split distinct-GAV Maven .pom siblings into separate packages

### PR DESCRIPTION
**Pull request by mstykow**
## Summary

- Maven assembly collapsed multiple INDEPENDENT `.pom` files in one directory — each with a distinct group:artifact:version (GAV) — into a single top-level package whose `datafile_paths` listed every pom. This happens in directories of standalone `.pom` fixtures (e.g. ant-ivy's `test/java/org/apache/ivy/plugins/parser/m2/` holds ~46) and in local-repo layouts.
- Add a conservative, Maven-specific guard in `src/assembly/sibling_merge.rs`: when two or more distinct Maven purls are present in a directory, emit one package per distinct purl instead of merging them all. Identity is keyed off the package `purl` (the GAV proxy).
- Purl-less supplementary datafiles (`pom.properties`, `META-INF/MANIFEST.MF`) are left to the default single-package merge path, so the real single-module case (one `pom.xml` + supplementary siblings, all the SAME GAV) stays byte-for-byte unchanged. The divergence only triggers at 2+ distinct purls.

## Issues

- Covers: Maven assembly over directories of distinct-GAV `.pom` files (found verifying apache/ant-ivy).

## Scope and exclusions

- Included: GAV-conflict guard in the Maven sibling merge; assembly unit tests, a scanner-wired contract test, and an assembly golden over a small synthetic multi-pom directory.
- Explicit exclusions: No change to `AssemblyMode` (Maven stays `SiblingMerge`, not switched to `OnePerPackageData`). No change to the scan-wide nested META-INF merge — `should_skip_nested_merge` already skips directories with more than one POM. No benchmark-row updates (recorded separately).

## How to verify

- `cargo test --lib assembly::assembly_test::tests::test_maven` and `cargo test --lib parsers::maven::scan_test` exercise the split and the same-GAV-merge-preserved cases.
- `cargo test --features golden-tests --test assembly_golden maven` covers the new `maven-distinct-gav-poms` golden plus the unchanged `maven-basic` / `maven-meta-inf-basic` single-module goldens.
- Edge case worth poking: a real `.m2` local-repo layout where each `.pom` lives in its own `<artifact>/<version>/` directory already assembled correctly via per-directory grouping (one purl per directory → default path); this fix targets the flat-directory case.

## Intentional differences from Python

- None. This brings Provenant in line with ScanCode, which emits one package per distinct-GAV `.pom`.

## Follow-up work

- None.

## Expected-output fixture changes

- Files changed: new `testdata/assembly-golden/maven-distinct-gav-poms/expected.json` (newly generated; not a modification of an existing fixture).
- Why the new expected output is correct: the three synthetic poms carry distinct GAVs (`org.example/alpha@1.0`, `org.example/beta@2.0`, `org.other/gamma@3.0`), so they must assemble into three separate packages, each owning its single `.pom` datafile. `gamma`'s declared dependency on `alpha` is correctly attributed to the `gamma` package. No existing maven golden changed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/assembly/sibling_merge.rs [hunk 1]
```diff
@@ -151,6 +155,99 @@ fn assemble_single_sibling_package(
...
+                .collect(),
+            None => Vec::new(),
+        };
+
+        results.push(build_directory_merge_output(
+            Some(package),
+            pending_dependencies,
+            vec![*idx],
+        ));
+    }
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/assembly/sibling_merge.rs]: Fixed in 61d0a3518b959d214ff4c0457e3a65d702b1bc15: the multi-GAV path now groups all datafiles sharing a purl and merges them into one package (attaching every datafile), so a duplicate-GAV file is no longer skipped/orphaned. Added unit test `test_maven_duplicate_gav_poms_merge_without_orphaning_files` covering exactly this case (two same-GAV poms + one distinct).

File: src/assembly/sibling_merge.rs [hunk 2]
```diff
@@ -151,6 +155,99 @@ fn assemble_single_sibling_package(
...
+/// standalone `.pom` fixtures or a local-repo layout where each `.pom` carries a
+/// different GAV. The default single-package sibling merge would collapse all of
+/// them into one top-level package; instead each distinct GAV must stay its own
+/// package.
+///
+/// Identity is keyed off the package `purl` (the GAV proxy). Datafiles with no
+/// purl (a `pom.properties` or `MANIFEST.MF` carrying only supplementary,
+/// purl-less metadata) are left to the default merge path: this divergence only
+/// triggers when two or more distinct Maven purls are present in the directory,
+/// so the single-module case is byte-for-byte unchanged.
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/assembly/sibling_merge.rs]: Fixed in 61d0a3518b959d214ff4c0457e3a65d702b1bc15: reworded the doc-comment to say purl-less supplementary files are not attached in the multi-GAV path and that the function returns `None` (falling through to the default merge) only when fewer than two distinct purls are present.

File: src/assembly/sibling_merge.rs [hunk 3]
```diff
@@ -151,6 +155,99 @@ fn assemble_single_sibling_package(
...
+            vec![*idx],
+        ));
+    }
+
+    Some(results)
+}
+
+fn is_maven_sibling_config(config: &AssemblerConfig) -> bool {
+    config.datasource_ids.contains(&DatasourceId::MavenPom)
+}
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/assembly/sibling_merge.rs:286]: Fixed in 61d0a3518b959d214ff4c0457e3a65d702b1bc15: `is_maven_sibling_config` now requires both `MavenPom` and `MavenPomProperties`, binding the guard to that one config.
